src/toil/provisioners/gceProvisioner.py

```diff
@@ -0,0 +1,915 @@
+            logger.error("Failed to launch %d workers", len(instancesLaunched)-workersCreated)
+        return workersCreated
+
+    def getProvisionedWorkers(self, nodeType, preemptable):
+        entireCluster = self._getNodesInCluster(nodeType=nodeType)
+        logger.debug('All nodes in cluster: %s', entireCluster)
+        workerInstances = [i for i in entireCluster if i.private_ips[0] != self.leaderIP]
+        logger.debug('All workers found in cluster: %s', workerInstances)
+        # TODO: get spot workers
+        #workerInstances = [i for i in workerInstances if preemptable != (i.spot_instance_request_id is None)]
+        #logger.debug('%spreemptable workers found in cluster: %s', 'non-' if not preemptable else '', workerInstances)
+        return [Node(publicIP=i.public_ips[0], privateIP=i.private_ips[0],
+                     name=i.name, launchTime=i.created_at, nodeType=i.size,
+                     preemptable=preemptable)
+                for i in workerInstances]
+
+    def _getLeader(self):
+        instances = self._getNodesInCluster()
+        instances.sort(key=lambda x: x.created_at)
+        try:
+            leader = instances[0]  # assume leader was launched first
+        except IndexError:
+            raise NoSuchClusterException(self.clusterName)
+        return leader
+
+    def _getNodesInCluster(self, nodeType=None):
+        instanceGroup = self._getDriver().ex_get_instancegroup(self.clusterName, zone=self.zone)
+        instances = instanceGroup.list_instances()
+        if nodeType:
+            instances = [instance for instance in instances if instance.size == nodeType]
+        return instances
+
+    def _getDriver(self):
+        driverCls = get_driver(Provider.GCE)
+        return driverCls(self.clientEmail,
+                         self.googleJson,
+                         project=self.projectId,
+                         datacenter=self.zone)
+
+    @classmethod
+    def _copySshKeys(cls, instanceIP, keyName):
+        """ Copy authorized_keys file to the core user from the keyName user."""
+        if keyName == 'core':
+            return True
+
+        # Make sure that keys are there.
+        if not cls._waitForSSHKeys(instanceIP, keyName=keyName):
+            return False
+
+        # TODO: Check if there is another way to ssh to a GCE instance with Google credentials
+
+        # copy keys to core user so that the ssh calls will work
+        # - normal mechanism failed unless public key was in the google-ssh format
+        # - even so, the key wasn't copied correctly to the core account
+        keyFile = '/home/%s/.ssh/authorized_keys' % keyName
+        cls._sshInstance(instanceIP, '/usr/bin/sudo', '/usr/bin/cp', keyFile, '/home/core/.ssh', user=keyName)
+        cls._sshInstance(instanceIP, '/usr/bin/sudo', '/usr/bin/chown', 'core',
+                         '/home/core/.ssh/authorized_keys', user=keyName)
+
+        return True
+
+    def _waitForNode(self, instanceIP, role):
```

Comment:
This should return which part failed instead of `False`.  That way a more useful error message than `raise RuntimeError("Failed to start leader")` is returned.

Reply:
Where do you see this error? That is not the only message.